src/test/java/seedu/exchangecoursemapper/FilterCoursesCommandTest.java

```diff
@@ -28,19 +29,66 @@ public void setUp() {
     }
 
     @Test
-    public void getNusCourseCode_inputWithCourseCode_expectNusCourseCode() throws ExchangeCourseMapperException {
-        String userInput = "filter cs3244";
-        String nusCourseCode = filterCoursesCommand.getNusCourseCode(userInput);
-        assertEquals("cs3244", nusCourseCode);
+    public void parseFilterCommand_inputWithOneCourse_expectSeparatedInput() {
+        String userInput = "filter cs3241";
+        String[] descriptionSubstrings = filterCoursesCommand.parseFilterCommand(userInput);
+        assertArrayEquals(new String[]{"filter", "cs3241"}, descriptionSubstrings);
     }
 
     @Test
-    public void getNusCourseCode_inputWithNoCourseCode_expectException() {
-        String userInput = "filter";
+    public void parseFilterCommand_inputWithTwoCourses_expectException() {
+        String userInput = "filter ee2026 cs3241";
         IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> {
-            filterCoursesCommand.getNusCourseCode(userInput);
+            filterCoursesCommand.parseFilterCommand(userInput);
         });
-        assertEquals("Please provide the course code you would like to search for.", e.getMessage());
+        assertEquals("Please note that we can only filter for only one NUS Course!",
+                e.getMessage());
+    }
+
+    @Test
+    public void parseDeleteCommand_inputWithNoIndexes_expectException() {
+        String userInput = "filter ";
+        IllegalArgumentException e = assertThrows(IllegalArgumentException.class, () -> {
+            filterCoursesCommand.parseFilterCommand(userInput);
+        });
+        assertEquals("Please provide the course code you would like to search for.",
+                e.getMessage());
+    }
+
+    @Test
+    public void isValidSocCourseCode_inputWithCsCourseCode_expectTrue() {
+        String userInput = "cs3244";
+        boolean isValidSocCourseCode = filterCoursesCommand.isValidSocCourseCode(userInput);
+        assertTrue(isValidSocCourseCode);
+    }
+
+    @Test
+    public void isValidSocCourseCode_inputWithEeCourseCode_expectTrue() {
+        String userInput = "ee2026";
+        boolean isValidSocCourseCode = filterCoursesCommand.isValidSocCourseCode(userInput);
+        assertTrue(isValidSocCourseCode);
+    }
+
+    @Test
+    public void isValidSocCourseCode_inputWithBtCourseCode_expectTrue() {
```

Comment:
Could this be repeated testing as the outcome is the same?

Reply:
For this, I was thinking to test if the code works for every SoC coded module! But I agree it might be repeated testing, hence I would change this part to a test for a non SoC coded module!